download-latest-release.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
```

Comment:
Docstrings aren't a thing in bash, but I think a short comment explaining what the script does would be useful to other devs.

```suggestion
#!/bin/bash

# Downloads the Debian tarball from the latest Synapse release on GitHub
# and extracts it into debian/incoming/.
```